Source/WebCore/editing/cocoa/DataDetection.h

```diff
@@ -63,7 +69,8 @@ class DataDetection {
 #if PLATFORM(MAC)
     WEBCORE_EXPORT static std::optional<DetectedItem> detectItemAroundHitTestResult(const HitTestResult&);
 #endif
-    WEBCORE_EXPORT static NSArray *detectContentInRange(const SimpleRange&, OptionSet<DataDetectorType>, std::optional<double> referenceDate);
+    WEBCORE_EXPORT static void detectContentInFrame(const WeakPtr<LocalFrame>&, OptionSet<DataDetectorType>, std::optional<double>, CompletionHandler<void(NSArray *)>&&);
```

Comment:
I think it's fine for parameters to use raw pointers, but let's see if @cdumez has something to say.

Reply:
Yes, fine to pass parameters as raw pointers. Of course, this assumes that the call site has a strong ref to this pointer on the stack before passing it in argument.